src/Raven.Client/Documents/AI/Chat.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Operations;
+using Raven.Client.Documents.Operations.AI.Agents;
+using Raven.Client.Util;
+
+namespace Raven.Client.Documents.AI;
+internal class Chat<T> : IChatOperations<T> where T : new()
+{
+    private readonly AiOperations _aiOperations;
+    private readonly string _agent;
+    private readonly Dictionary<string, object> _scope;
+
+    private string _chatId;
+    private AiUsage _totalUsage;
+    private List<ToolRequest> _toolsRequests;
+    private List<ToolResponse> _toolsResponses = [];
+    private string _userPrompt;
+
+    private bool _firstRun = true;
+    public Chat(AiOperations aiOperations, string agent, Dictionary<string, object> scope)
+    {
+        _aiOperations = aiOperations;
+        _agent = agent;
+        _scope = scope;
+    }
+    public Chat(AiOperations aiOperations, string chatId)
+    {
+        _aiOperations = aiOperations;
+        _chatId = chatId;
+    }
+
+    public IEnumerable<ToolRequest> OpenTools() => _toolsRequests ?? throw new InvalidOperationException("You have to call RunAsync first");
```

Comment:
Do we want to call it `OpenTools`? Can we find a better term? 

`PendingActions` - maybe, that match better the term in the agent description?